src/ui/manage-ui/src/FamilyHubs.ServiceDirectory.Admin.Web/Pages/Welcome.cshtml.cs

```diff
@@ -39,48 +40,70 @@ public WelcomeModel(
     public async Task OnGet()
     {
         var familyHubsUser = HttpContext.GetFamilyHubsUser();
-        SetVisibleSections(familyHubsUser.Role);
-
-        Heading = familyHubsUser.FullName;
-        //todo: no magic strings
-        if (familyHubsUser.OrganisationId == "-1")
-        {
-            SubHeading = "Department for Education";
-        }
-        else
-        {
-            if (long.TryParse(familyHubsUser.OrganisationId, out var organisationId))
-            {
-                //todo: looks like we get the organisation with *all* it's services, just so that we can use the name!
-                var organisation = await _serviceDirectoryClient.GetOrganisationById(organisationId);
-                SubHeading = organisation.Name;
-            }
-        }
+        await SetModelPropertiesBasedOnRole(familyHubsUser);
 
         await _cacheService.ResetLastPageName();
     }
 
-    private void SetVisibleSections(string role)
+    private async Task SetModelPropertiesBasedOnRole(FamilyHubsUser familyHubsUser)
     {
-        switch (role)
+        switch (familyHubsUser.Role)
         {
             case RoleTypes.DfeAdmin:
-                MenuPage = MenuPage.Dfe;
+                await SetDfeAdminProperties(familyHubsUser);
                 break;
-
-            case RoleTypes.LaDualRole:
             case RoleTypes.LaManager:
-                MenuPage = MenuPage.La;
+            case RoleTypes.LaDualRole:
+                await SetLaAdminProperties(familyHubsUser);
                 break;
-
-            case RoleTypes.VcsDualRole:
             case RoleTypes.VcsManager:
-                MenuPage = MenuPage.Vcs;
+            case RoleTypes.VcsDualRole:
+                await SetVcsAdminProperties(familyHubsUser);
                 break;
-
             default:
-                throw new InvalidOperationException($"Unknown role: {role}");
+                throw new InvalidOperationException($"Unknown role: {familyHubsUser.Role}");
+        }
+    }
+    
+    private async Task SetDfeAdminProperties(FamilyHubsUser familyHubsUser)
+    {
+        Heading = familyHubsUser.FullName;
+        CaptionText = await GetOrganisationName(familyHubsUser);
+        Description = "Manage users, services, locations and organisations and view performance data.";
```

Comment:
This text is repeated twice more - perhaps use a constant

Reply:
I've updated the code to remove duplication. Thanks